go/carmen/memory_footprint.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package carmen
+
+import "github.com/Fantom-foundation/Carmen/go/common"
+
+type MemoryFootprint interface {
+	// GetChild returns a child of the memory footprint with the given name.
+	GetChild(name string) MemoryFootprint
```

Comment:
I guess this method should be removed, right? It is still hard to use when the list of names is not known.